### PR TITLE
Update go pkg versions

### PR DESCRIPTION
**Pull request by lkysow**
* controller-runtime => v0.9.0
* client-go => v0.21.1
* remove darwin 386 build because it doesn't work with Go 1.16 and we don't release a Docker image for this arch anyway

This fixes the vulnerability for golang.org/x/crypto.

### CODE REVIEW

**Comment by lkysow** [comment]
```
trivy fs /Users/lkysow/go/bin/consul-k8s
2021-06-14T08:56:41.842-0700	WARN	OS is not detected and vulnerabilities in OS packages are not detected.
2021-06-14T08:56:41.842-0700	INFO	Detecting gobinary vulnerabilities...

.
=
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```

**Review by kschoche — APPROVED** [review]
Nice!

File: .circleci/config.yml [hunk 1]
```diff
@@ -160,7 +160,7 @@ workflows:
             - go-fmt-and-vet
             - linters
       - build-distro:
-          OS: "darwin freebsd linux windows"
```
Comment (ishustava) [inline .circleci/config.yml:163]: why are we removing the darwin build?

Comment (lkysow) [reply .circleci/config.yml:163]: Sorry, didn't update PR comment.

* we don't release docker images for this arch
* breaks on golang 1.16 (https://golang.org/doc/go1.16#386)

Comment (lkysow) [reply .circleci/config.yml:163]: See https://app.circleci.com/pipelines/github/hashicorp/consul-k8s/2088/workflows/35dc4d70-b494-4ae7-9e65-2006c386ace7/jobs/11104

Comment (ishustava) [reply .circleci/config.yml:163]: Ah I see. Yep, and it looks like we [don't release](https://releases.hashicorp.com/consul-k8s/0.26.0-beta3/) darwin/386 binary either.

**Review by ishustava — APPROVED** [review]

**Pull request merged**